tools/test/toolchains/test_toolchains.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Test the arm toolchain."""
+import pytest
+import os
+import sys
+from unittest import TestCase
+from collections import OrderedDict
+
+import mock
+
+
+ROOT = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "..", "..")
+)
+sys.path.insert(0, ROOT)
+
+from tools.toolchains.arm import ARM_STD, ARM_MICRO, ARMC6
+from tools.toolchains.gcc import GCC_ARM
+from tools.toolchains.iar import IAR
+from tools.utils import NotSupportedException
+
+class TestArmToolchain(TestCase):
+    """Test Arm classes."""
+
+    def test_arm_default_lib(self):
+        """Test that linker flags are correctly added to an instance of ARM."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.__dict__["supported_c_lib"] = OrderedDict([(u'arm_c_lib', [u'small'])])
```

Comment:
It is not usually recommended to use `__` members. 
Was it not possible to use `mock_target.supported_c_lib = { "arm_c_lib" : ["small"]}` ?

Reply:
__dict__ is syntax from python 
`https://codesachin.wordpress.com/2016/06/09/the-magic-behind-attribute-access-in-python/` and 
mock_target.supported_c_lib = { "arm_c_lib" : ["small"]} was not working.

Reply:
Can you make sure that the test can be run (see https://github.com/rajkan01/mbed-os/pull/4#issuecomment-559810110)  and I will try it?

Reply:
I just tried `mock_target.supported_c_lib = {'arm_c_lib': 'small'}` and the test passed fine. Please update it.